third_party/webhook/authz.go

```diff
@@ -89,36 +91,42 @@ func (a *authorizer) clientX509(ctx context.Context) (*client, error) {
 }
 
 // getSubjectAccessReview extracts the subject access review object from the request and returns it.
-func (a *authorizer) getSubjectAccessReview(ctx context.Context, req *http.Request) (*authz.SubjectAccessReview, error) {
+func (a *authorizer) getSubjectAccessReview(ctx context.Context, req *http.Request) (bool, *authz.SubjectAccessReview, error) {
+	isV1Beta := false // TODO: Remove me! Temporary fix to support both v1 and v1beta1 versions of SubjectAccessReview.
 	b, err := ioutil.ReadAll(req.Body)
 	if err != nil {
-		return nil, fmt.Errorf("body read error for authorization request, %v", err)
+		return isV1Beta, nil, fmt.Errorf("body read error for authorization request, %v", err)
 	}
 	if len(b) == 0 {
-		return nil, fmt.Errorf("empty body for authorization request")
+		return isV1Beta, nil, fmt.Errorf("empty body for authorization request")
 	}
 	if isLogEnabled(ctx, LogTraceServer) {
 		getLogger(ctx).Printf("request body: %s\n", b)
 	}
 	var r authz.SubjectAccessReview
 	if err := json.Unmarshal(b, &r); err != nil {
-		return nil, fmt.Errorf("invalid JSON request '%s', %v", b, err)
+		return isV1Beta, nil, fmt.Errorf("invalid JSON request '%s', %v", b, err)
 	}
-	// TODO: This is a temporary fix to support both v1 and v1beta1 versions of SubjectAccessReview & will be removed in future.
+	// TODO: Remove me! This is a temporary fix to support both v1 and v1beta1 versions of SubjectAccessReview & will be removed in future.
 	if r.APIVersion == authzSupportedBetaVersion {
-		// This is feasible as the only difference between v1 and v1beta1 is the APIVersion.
-		r.APIVersion = authzSupportedVersion
+		isV1Beta = true
+		var rV1Beta1 authzv1beta1.SubjectAccessReview
+		if err := json.Unmarshal(b, &rV1Beta1); err != nil {
+			return isV1Beta, nil, fmt.Errorf("invalid JSON request '%s', %v", b, err)
+		}
+		r = ConvertIntoV1(rV1Beta1)
+		glg.Warn("Your cluster is using deprecated authorization.k8s.io/v1beta1 instead of authorization.k8s.io/v1", "convertedFrom:", rV1Beta1, "convertedTo:", r)
```

Comment:
```suggestion
```
This will create a lot of warning log and affect the performance.
I think we do not really need this log. If we need that, please limit the output rate of the log (maybe once an hour).

Reply:
Nice point. Let me comment this out for future usage and will delete this for Garm v4, that only supports v1, not v1beta1

Reply:
handled.